src/main/java/io/confluent/connect/jdbc/JdbcSourceConnectorConfig.java

```diff
@@ -107,6 +107,12 @@
   public static final String TABLE_BLACKLIST_DEFAULT = "";
   private static final String TABLE_BLACKLIST_DISPLAY = "";
 
+  public static final String INCREMENTING_COLUMN_USE_PRIMARY_KEY_CONFIG = "incrementing.column.use.primary.key";
```

Comment:
The reason I factored it this way is because the `mode` configuration seems orthogonal to the concern about what determines which column is the incrementing column. (Previously, there were two options for determining which column is the incrementing column: assume anything that is AUTO INCREMENT is the incrementing column, or specify a fixed incrementing column name.)

Ie, if we were to roll this into the `mode` config, you'd technically need two new options: `timestamp+incrementing-primary-key` and `incrementing-primary-key` -- this isn't terrible but if another mode were introduced that was in addition to `incrementing` and `incrementing+timestamp` that was incrementing column-dependent, you'd need two new `mode`s, etc.